### PR TITLE
chore: addopt bandit static linter

### PR DESCRIPTION
**Pull request by bruno-fs**
Yet another linter enabled in ruff. This time, [bandit](https://pypi.org/project/bandit/).

The inspiration to add this check came from this comment: https://github.com/quipucords/quipucords/pull/2492#discussion_r1345776403

### CODE REVIEW

**Review by infinitewarp — APPROVED** [review]
big 👏👏👏 for removing `assert`s from runtime code

File: pyproject.toml [hunk 1]
```diff
@@ -112,9 +113,14 @@ known-first-party = [
 
 [tool.ruff.per-file-ignores]
 "__init__.py" = ["F401", "D104"]
-"**/test_*.py" = [
+"**test**" = [
```
Comment (abellotti) [inline pyproject.toml:116]: seems to be much more relaxed, what weren't we catching with **/test_*.py ?

Comment (bruno-fs) [reply pyproject.toml:116]: I changed to also capture `conftest.py`.

**Comment by bruno-fs** [comment]
> big 👏👏👏 for removing asserts from runtime code

@infinitewarp thank @mirekdlugosz for calling it out in another PR ;D

File: pyproject.toml (hunk 1 above)
Comment (abellotti) [reply pyproject.toml:116]: Can we do this in Ruff,   "**/test_*.py", "conftest.py" = [ .... ] so we don't duplicate the exceptions?

Comment (bruno-fs) [reply pyproject.toml:116]: I did try something similar and now that. no luck. Will check the docs tho, maybe there's a way 🤔

**Pull request merged**